formats/config/common.go

```diff
@@ -0,0 +1,147 @@
+// Package config implements decoding, encoding and
+// manipulation git config files.
+//
+// Reference: https://git-scm.com/docs/git-config
+package config
+
+import "strings"
+
+// New creates a new config instance.
+func New() *Config {
+	return &Config{}
+}
+
+type Config struct {
+	Comment  *Comment
+	Sections []*Section
+	Includes []*Include
+}
+
+type Section struct {
+	Name        string
+	Options     []*Option
+	Subsections []*Subsection
+}
+
+type Subsection struct {
+	Name    string
+	Options []*Option
+}
+
+type Option struct {
+	// Key preserving original caseness.
+	// Use IsKey instead to compare key regardless of caseness.
+	Key   string
+	// Original value as string, could be not notmalized.
+	Value string
+}
+
+// A reference to a included configuration.
+type Include struct {
+	Path   string
+	Config *Config
+}
+
+type Comment string
+
+// IsKey returns true if the given key matches
+// this options' key in a case-insensitive comparison.
+func (o *Option) IsKey(key string) bool {
+	return strings.ToLower(o.Key) == strings.ToLower(key)
+}
+
+func (c *Config) Section(name string) *Section {
+	for i := len(c.Sections) - 1; i >= 0; i-- {
```

Comment:
Does it has to do with disambiguating sections with the same name to the one that was appended last?, is name supposed to be an unique id?